pgjdbc/src/main/java/org/postgresql/largeobject/BlobInputStream.java

```diff
@@ -76,43 +77,131 @@ public BlobInputStream(LargeObject lo, int bsize) {
    */
   public BlobInputStream(LargeObject lo, int bsize, long limit) {
     this.lo = lo;
-    buffer = null;
-    bpos = 0;
-    apos = 0;
-    this.bsize = bsize;
-    this.limit = limit;
+    this.bufferSize = bsize;
+    // Treat -1 as no limit for backward compatibility
+    this.limit = limit == -1 ? Long.MAX_VALUE : limit;
   }
 
   /**
    * The minimum required to implement input stream.
    */
   public int read() throws java.io.IOException {
-    LargeObject lo = getLo();
-    try {
-      if (limit > 0 && apos >= limit) {
```

Comment:
trying to remember why we would check for a limit >0 here ?

Reply:
Previously, `-1` was there to mean "no limit".
Now I use `Long.MAX_VALUE` so we do not need the old `limit > 0`, and `actualPosition >= limit` is enough.